x-pack/plugin/esql/qa/testFixtures/src/main/resources/topN.csv-spec

```diff
@@ -134,3 +134,37 @@ Otmar             |Herbst           |[-8.19, -1.9, -0.32]        |[-0.32, -1.90,
 null              |Swan             |-8.46                       |-8.46                        |-8                   |10034          
 Sanjiv            |Zschoche         |[-7.67, -3.25]              |[-3.25, -7.67]               |[-3, -8]             |10053
 ;
+
+lookupBeforeSort
+FROM employees
+| WHERE emp_no < 10005
+| RENAME languages AS int
+| LOOKUP int_number_names ON int
+| RENAME name AS languages
+| KEEP languages, emp_no
+| SORT languages ASC, emp_no ASC
+;
+
+languages:keyword | emp_no:integer
+             five | 10002
+             five | 10004
+             four | 10003
+              two | 10001
+;
+
+lookupAfterSort
+FROM employees
+| WHERE emp_no < 10005
+| SORT languages ASC, emp_no ASC
+| RENAME languages AS int
+| LOOKUP int_number_names ON int
```

Comment:
I can't seem to find any csv tests with `| LOOKUP in_number_names ON int WITH ... AS ...`, which I'd expect should work similarly to ENRICH. I think it's important that the columns can be given chosen names in the lookup command, and we should test this.

Reply:
I haven't implemented the renaming. I'm not 100% sure you need to be able replace because you normally have control over the table itself in the request. Except. I don't think everyone will. So it'll be important. It's important here because of the test making the table so general. But I'd imagine a table being defined by, say, a kibana plugin.

Reply:
We can do this in a follow-up, as long as we don't bake the assumption of always getting the attribute names from the table columns into the execution somehow.

I think it's important as renaming table columns may not always be possible/ergonomic + grok/dissect currently have a problem where you cannot rename the generated attributes and it causes complications in optimization rules. (The column attributes may shadow e.g. field attributes and then you have to awkwardly rename stuff - in terms of opt. rules that means introducing a projection which you may not want.)

Reply:
This'll be a followup.